src/core/net/ip6.cpp

```diff
@@ -374,6 +391,49 @@ otError Ip6::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, IpProto
     if (aMessageInfo.GetPeerAddr().IsMulticast() &&
         aMessageInfo.GetPeerAddr().GetScope() > Address::kRealmLocalScope)
     {
+        if (GetInstance().GetThreadNetif().GetMle().HasSleepyChildrenSubscribed(header.GetDestination()))
+        {
+            Message *messageCopy = NULL;
+
+            if ((messageCopy = aMessage.Clone()) != NULL)
+            {
+                otLogInfoIp6(GetInstance(), "Message copy for indirect transmission to sleepy children");
+
+                // compute checksum
+                checksum = ComputePseudoheaderChecksum(header.GetSource(), header.GetDestination(),
+                                                       payloadLength, aIpProto);
+
+                switch (aIpProto)
+                {
+                case kProtoUdp:
+                    error = mUdp.UpdateChecksum(*messageCopy, checksum);
```

Comment:
This seems to duplicate code from further down.  Can we move the checksum update code below further up to avoid this duplication?

Reply:
Thanks. Updated.